packages/ui/src/ui.tsx

```diff
@@ -0,0 +1,102 @@
+import * as ds from "@devicescript/core"
+import { Image, ImageContext } from "@devicescript/graphics"
+
+export abstract class BaseElement<T> implements JSX.Element {
+    constructor(props: T) {
+        Object.assign(this, props)
+    }
+    abstract render(ctx: ImageContext): void
+}
+
+export abstract class ParentElement<T> extends BaseElement<T> {
+    children?: JSX.Element | JSX.Element[]
+
+    renderChildren(ctx: ImageContext) {
+        if (!this.children) return
+        if (Array.isArray(this.children))
+            for (const ch of this.children)
+                ch.render(ctx)
+        else
+            this.children.render(ctx)
+    }
+}
+
+export class Translate extends ParentElement<{ x: number, y: number } & JSX.BaseProps> {
+    x: number
+    y: number
+
+    render(ctx: ImageContext): void {
+        ctx.save()
+        ctx.translate(this.x, this.y)
+        this.renderChildren(ctx)
+        ctx.restore()
+    }
+}
+
+export class Text extends BaseElement<{ children: string }> {
+    children: string
```

Comment:
what about optional x,y in the base element?

Reply:
My idea was to use `<Translate>` on elements to move them; this allows for `<Columns>`, `<Rows>` or similar in future that lays out elements (that would require some size computation going up the tree before-hands). But I guess for now having to use `<Translate>` everywhere is annoying?

Reply:
It can be annoying...